File: app/src/main/java/org/ole/planet/myplanet/ui/team/teamResource/AdapterTeamResource.java
package org.ole.planet.myplanet.ui.team.teamResource;

import android.content.Context;
import android.content.SharedPreferences;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import org.ole.planet.myplanet.callback.OnHomeItemClickListener;
import org.ole.planet.myplanet.databinding.RowTeamResourceBinding;
import org.ole.planet.myplanet.model.RealmMyLibrary;
import org.ole.planet.myplanet.model.RealmMyTeam;

import java.util.List;

import io.realm.Realm;

public class AdapterTeamResource extends RecyclerView.Adapter<AdapterTeamResource.ViewHolderTeamResource> {
    private RowTeamResourceBinding rowTeamResourceBinding;
    private Context context;
    private List<RealmMyLibrary> list;
    private Realm mRealm;
    private OnHomeItemClickListener listener;
    private SharedPreferences settings;
    private String teamCreator;

    public AdapterTeamResource(Context context, List<RealmMyLibrary> list, Realm mRealm, String teamId, SharedPreferences settings) {
        this.context = context;
        this.list = list;
        this.mRealm = mRealm;
        this.settings = settings;
        teamCreator = RealmMyTeam.getTeamCreator(teamId, mRealm);
        if (context instanceof OnHomeItemClickListener) {
            listener = (OnHomeItemClickListener) context;
        }
    }

    @NonNull
    @Override
    public ViewHolderTeamResource onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        rowTeamResourceBinding = RowTeamResourceBinding.inflate(LayoutInflater.from(context), parent, false);
        return new ViewHolderTeamResource(rowTeamResourceBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolderTeamResource holder, int position) {
        rowTeamResourceBinding.tvTitle.setText(list.get(position).getTitle());
        rowTeamResourceBinding.tvDescription.setText(list.get(position).getDescription());
        holder.itemView.setOnClickListener(v -> {
            if (listener != null) {
                listener.openLibraryDetailFragment(list.get(position));
            }
        });
        rowTeamResourceBinding.ivRemove.setOnClickListener(view -> {
        });
        if (!settings.getString("userId", "--").equalsIgnoreCase(teamCreator)) {
            rowTeamResourceBinding.ivRemove.setVisibility(View.GONE);
        }
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    static class ViewHolderTeamResource extends RecyclerView.ViewHolder {
        RowTeamResourceBinding rowTeamResourceBinding;

        public ViewHolderTeamResource(RowTeamResourceBinding rowTeamResourceBinding) {
            super(rowTeamResourceBinding.getRoot());
            this.rowTeamResourceBinding = rowTeamResourceBinding;
        }
    }
}

File: app/src/main/java/org/ole/planet/myplanet/ui/team/teamResource/AdapterTeamResource.kt
package org.ole.planet.myplanet.ui.team.teamResource

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import org.ole.planet.myplanet.callback.OnHomeItemClickListener
import org.ole.planet.myplanet.databinding.RowTeamResourceBinding
import org.ole.planet.myplanet.model.RealmMyLibrary
import org.ole.planet.myplanet.utilities.DiffUtils

class AdapterTeamResource(
    private val context: Context,
    private val canRemoveResources: Boolean,
    private val updateListener: ResourceUpdateListner,
    private val onRemoveResource: (RealmMyLibrary, Int) -> Unit,
) : ListAdapter<RealmMyLibrary, AdapterTeamResource.ViewHolderTeamResource>(ITEM_CALLBACK) {

    private var listener: OnHomeItemClickListener? = null

    init {
        if (context is OnHomeItemClickListener) {
            listener = context
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolderTeamResource {
        val binding = RowTeamResourceBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolderTeamResource(binding)
    }

    override fun onBindViewHolder(holder: ViewHolderTeamResource, position: Int) {
        val adapterPosition = holder.bindingAdapterPosition
        if (adapterPosition == RecyclerView.NO_POSITION) return

        val resource = getItem(adapterPosition)

        holder.binding.apply {
            tvTitle.text = resource.title
            tvDescription.text = resource.description

            root.setOnClickListener {
                listener?.openLibraryDetailFragment(resource)
            }

            ivRemove.apply {
                visibility = if (canRemoveResources) View.VISIBLE else View.GONE
                setOnClickListener {
                    val currentPosition = holder.bindingAdapterPosition
                    if (currentPosition != RecyclerView.NO_POSITION) {
                        onRemoveResource(getItem(currentPosition), currentPosition)
                    }
                }
            }
        }
    }

    fun removeResourceAt(position: Int) {
        if (position < 0 || position >= currentList.size) return
        val newList = currentList.toMutableList()
        newList.removeAt(position)
        submitList(newList)
        updateListener.onResourceListUpdated()
    }

    class ViewHolderTeamResource(val binding: RowTeamResourceBinding) : RecyclerView.ViewHolder(binding.root)

    companion object {
        private val ITEM_CALLBACK = DiffUtils.itemCallback<RealmMyLibrary>(
            areItemsTheSame = { oldItem, newItem -> oldItem.id == newItem.id },
            areContentsTheSame = { oldItem, newItem -> oldItem.title == newItem.title && oldItem.description == newItem.description }
        )
    }
}
